File: 39-CombinationSum/39-CombinationSum.cpp
// Last updated: 6/11/2026, 11:38:02 AM
class Solution {
public:
    vector<vector<int>> ans;
    void rec(int ind, vector<int> temp, vector<int>& candidates, int target){
        if(ind == candidates.size()){
            if(target == 0){
                ans.push_back(temp);
                return;
            }
            return;
        }
        if(target<0) return;
        temp.push_back(candidates[ind]);
        rec(ind, temp, candidates, target-candidates[ind]);
        temp.pop_back();
        rec(ind+1, temp,candidates, target);
    }


    vector<vector<int>> combinationSum(vector<int>& candidates, int target) {
        ans.clear();
        vector<int> temp;
        rec(0, temp, candidates, target);
        return ans;
    }
};

File: 39-CombinationSum/39-CombinationSum.py
# Last updated: 6/11/2026, 11:38:08 AM
class Solution:
    def combinationSum(self, candidates: List[int], target: int) -> List[List[int]]:
        ans = []
        def combination(ind,temp,rsum,ans):
            if(rsum == 0):
                ans.append(temp.copy())
                return
            if(rsum<0 or ind<0):
                return
            temp.append(candidates[ind])
            combination(ind,temp,rsum-candidates[ind],ans)
            temp.pop()
            combination(ind-1,temp,rsum,ans)
        combination(len(candidates)-1,[],target,ans)
        return ans
            
                
                